File: 01/two.nim
import std/strutils
import sequtils

var
  ups = -1
  prev = -1
  total: int

let
  all_lines = "input.txt".lines.to_seq

for i, line in all_lines.pairs:
  try:
    total = line.parse_int +
      all_lines[i + 1].parse_int +
      all_lines[i + 2].parse_int
  except IndexDefect:
    ups.echo
    break
  if total > prev:
    ups.inc
  prev = total

File: 01/two.py
#!/bin/env python3

ups = 0

with open('input.txt') as infile:
    a, b, c = (
        int(next(infile).strip()),
        int(next(infile).strip()),
        int(next(infile).strip())
    )
    prev = a + b + c
    while True:
        try:
            a, b, c = b, c, int(next(infile).strip())
        except StopIteration:
            print(ups)
            break
        else:
            total = a + b + c
            if total > prev:
                ups += 1
            prev = total
